Support resizing and cropping native images

kevinsawicki opened the pull request:
This pull requests add cropping and resizing support to native images.

- [x] Add `resize` method
- [x] Add `crop` method
- [x] Add `getAspectRatio` method 
- [x] Add tests
- [x] Add docs

Closes #6523

davej commented:
Awesome. 👍

(Something small I noticed… the type signature for `nativeImage.getSize()` is incorrect, it should be an Object. Do I just PR the english docs to fix it or are the type signatures now being generated from somewhere?)

kevinsawicki commented:
> Do I just PR the english docs to fix it or are the type signatures now being generated from somewhere?

PR to `docs/api/native-image.md` should be good 👍 Thanks.

zcbenz commented on docs/api/native-image.md:
```diff
@@ -215,4 +215,28 @@ Marks the image as a template image.
 
+#### `image.crop(region)`
+
+* `region` Object - The region of the image to crop
```
`rect` would be a better name, `region` is usually an union of `rect`s.

zcbenz commented:
👍

davej commented:
Just a thought… if I do an image resize with one param:

```js
image.resize({ width: 200 });
```

Then I would have probably expected the `height` to scale proportionately with the `width` (rather than stay the same). It's rare that you would actually want to skew the image, the more common case is that you want to scale the image to a fixed width or height. Of course this can be done in userland using javascript and `image.getSize()` so it's not a big issue. Might be worth mentioning it in the docs though.

kevinsawicki commented:
> Then I would have probably expected the height to scale proportionately with the width (rather than stay the same)

Good call, I've updated this pull request to now have that behavior. Setting only `height` or `width` will use a preserved aspect ratio value for the other.

Also added a `getAspectRatio` API since that might be useful as well when resizing/cropping.

zcbenz commented:
👍
